### PR TITLE
S3 docs

### PR DESCRIPTION
**Pull request by csansoon**
## Describe your changes

DOCS!

## Checklist before requesting a review
- [x] I have updated the documentation if necessary

### CODE REVIEW

File: docs/sources/advanced/s3-storage.mdx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+Although your Latitude project is configured to materialize on S3, you will still need to configure your DuckDB source to read materialized queries from S3. This can be done by adding your S3 credentials to the DuckDB source configuration.
+
+```yaml
+type: duckdb
+details:
+  s3:
+    accessKeyId: <your-access-key-id>
+    secretAccessKey: <your-secret-access-key>
+    region: <your-region>
```
Comment (andresgutgon) [inline docs/sources/advanced/s3-storage.mdx:129]: Same

File: packages/storage_driver/src/drivers/s3/index.ts [hunk 2]
```diff
@@ -132,6 +134,23 @@ export class S3Driver extends StorageDriver {
...
+    )
+    await this.delete(from)
+  }
+
+  async delete(filepath: string): Promise<void> {
+    await this.client.send(
+      new DeleteObjectCommand({ Bucket: this.bucket, Key: filepath }),
+    )
+  }
+
```
Comment (andresgutgon) [inline packages/storage_driver/src/drivers/s3/index.ts:153]: this is from other PR no?

Comment (csansoon) [reply packages/storage_driver/src/drivers/s3/index.ts:153]: oh, yep. Chose the wrong base branch

File: docs/sources/advanced/s3-storage.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+Add your IAM user credentials to your Latitude project by modifying the `latitude.json` file:
+
+```json
+{
+    "name": <latitude-project-name>,
+    "version": <latitude-project-version>,
+    "storage": {
+        "type": "s3",
+        "bucket": <your-s3-bucket-name>,
+        "region": <your-s3-bucket-region>,
```
[the remark this answers is not in the document]
Comment (csansoon) [reply docs/sources/advanced/s3-storage.mdx:106]: I recommended it using a `Note` underneath the code.

[the remark this answers is not in the document]
Comment (andresgutgon) [reply docs/sources/advanced/s3-storage.mdx:106]: I see but what's the point recommending here the non-secure way

[the remark this answers is not in the document]
Comment (csansoon) [reply docs/sources/advanced/s3-storage.mdx:106]: Testing, for example. That's what I have been using

[the remark this answers is not in the document]
Comment (andresgutgon) [reply docs/sources/advanced/s3-storage.mdx:106]: Yes, but this is dangerous. You can test by setting the values in `.env` vars when we have all the pieces. Docs should not encourage storing secrets in git. People can commit it by mistake

**Pull request merged**